Tests/IGListBatchUpdateDataTests.m

```diff
@@ -123,14 +123,28 @@ - (void)test_whenMovingSections_withMoveFromConflictWithDelete_thatResultDropsTh
     IGListBatchUpdateData *result = [[IGListBatchUpdateData alloc] initWithInsertSections:indexSet(@[])
                                                                            deleteSections:indexSet(@[@2])
                                                                              moveSections:[NSSet setWithArray:@[newMove(2, 6), newMove(0, 2)]]
-                                                                         insertIndexPaths:[NSArray new]
-                                                                         deleteIndexPaths:[NSArray new]
-                                                                           moveIndexPaths:[NSArray new]];
+                                                                         insertIndexPaths:@[]
+                                                                         deleteIndexPaths:@[]
+                                                                           moveIndexPaths:@[]];
     XCTAssertEqual(result.deleteSections.count, 1);
     XCTAssertEqual(result.moveSections.count, 1);
     XCTAssertEqual(result.insertSections.count, 0);
     XCTAssertEqualObjects(result.deleteSections, indexSet(@[@2]));
     XCTAssertEqualObjects(result.moveSections.anyObject, newMove(0, 2));
 }
 
+- (void)test_whenDuplicateDeleteIndexPaths_thatDeletesStaggered {
+    IGListBatchUpdateData *result = [[IGListBatchUpdateData alloc] initWithInsertSections:indexSet(@[])
+                                                                           deleteSections:indexSet(@[])
+                                                                             moveSections:[NSSet new]
+                                                                         insertIndexPaths:@[]
+                                                                         deleteIndexPaths:@[newPath(2, 0), newPath(2, 0), newPath(2, 1), newPath(2, 4)]
+                                                                           moveIndexPaths:@[]];
+    XCTAssertEqual(result.deleteIndexPaths.count, 4);
+
+    NSArray *expected = @[newPath(2, 0), newPath(2, 1), newPath(2, 3), newPath(2, 4)];
```

Comment:
Made me double-check how this works. There's a bug that if you delete the _last_ item twice, it will throw b/c its deleting an item that doesn't exist...